src/blog/v8-release-78.md

```diff
@@ -0,0 +1,130 @@
+
+```js
+const {x, y} = object;
+```
+
+...is almost equivalent to the desugared form...
+
+```js
+const x = object.x;
+const y = object.y;
+```
+
+...except that it also needs to throw a special error for `object` being `undefined` or `null`...
+
+```
+$ v8 -e 'const object = undefined; const {x, y} = object;'
+unnamed:1: TypeError: Cannot destructure property `x` of 'undefined' or 'null'.
+const object = undefined; const {x, y} = object;
+                                 ^
+```
+
+...rather than the normal error you’d get when trying to dereference undefined:
+
+```
+$ v8 -e 'const object = undefined; object.x'
+unnamed:1: TypeError: Cannot read property 'x' of undefined
+const object = undefined; object.x
+                                 ^
+```
+
+This extra check made destructuring slower than simple variable assignment, as [reported to us via Twitter](https://twitter.com/mkubilayk/status/1166360933087752197).
+
+As of V8 v7.8, object destructuring is **as fast** as the equivalent desugared variable assignment (in fact, we generate the same bytecode for both). Now, instead of explicit `undefined`/`null` checks, we rely on an exception being thrown when loading `object.x`, and we intercept the exception if it’s the result of destructuring.
+
+### Lazy source positions
+
+When compiling bytecode from JavaScript, source position tables are generated that tie bytecode sequences to character positions within the source code. However, this information is only used when symbolizing exceptions or performing developer tasks such as debugging and profiling and so this is largely wasted memory.
+To avoid this, we now compile bytecode without collecting source positions (assuming no debugger or profiler is attached). The source positions are only collected when a stack trace is actually generated, for instance when calling `Error.stack` or printing an exception’s stack trace to the console. This does have some cost, as generating source positions requires the function to be reparsed and compiled, however most websites don’t symbolize stack traces in production and therefore don’t see any observable performance impact. In our lab testing we saw between 1-2.5% reductions in V8’s memory usage.
+
+<figure>
+  <img src="/_img/v8-release-78/memory-savings.svg" width="699" height="432" alt="" loading="lazy">
+  <figcaption>Memory savings from lazy source positions on an AndroidGo device</figcaption>
+</figure>
+
+### Faster RegExp match failures
+
+Generally, a RegExp attempts to find a match by iterating forward through the input string and checking for a match starting from each position. Once that position gets close enough to the end of the string that no match is possible, V8 now (in most cases) stops trying to find possible beginnings of new matches, and instead quickly returns a failure. This optimization applies to both compiled and interpreted regular expressions, and yields a speedup on workloads where
+failure to find a match is common, and
```

Comment:
Quick tip for next time: these hard line breaks shouldn’t be here. They become `<br>` which breaks the text flow depending on the viewport size.

Reply:
Thanks, missed this one.